### PR TITLE
stm32h5 supports  counter Timer and RTC

### PR DESCRIPTION
**Pull request by FRASTM**
Add the support of the Timers  and RTC to the stm32H5 serie.

Adapt the RTC counter driver to the stm32H5 serie
Enable PWM 2 and PWM 3 on the stm32h573i_dk board
Enable RTC sourced by the LSE clock on the stm32h573i_dk board (also works with LSI source clock)

Signed-off-by: Francois Ramu [francois.ramu@st.com](mailto:francois.ramu@st.com)

### CODE REVIEW

**Comment by FRASTM** [comment]
Running the west build -p auto -b stm32h573i_dk tests/drivers/pwm/pwm_api/
```
TESTSUITE pwm_basic succeeded                                                   
                                                                                
------ TESTSUITE SUMMARY START ------                                           
                                                                                
SUITE PASS - 100.00% [pwm_basic]: pass = 2, fail = 0, skip = 0, total = 2 durats
 - PASS - [pwm_basic.test_pwm_cycle] duration = 3.011 seconds                   
 - PASS - [pwm_basic.test_pwm_nsec] duration = 3.012 seconds                    
                                                                                
------ TESTSUITE SUMMARY END ------ 
```

**Comment by FRASTM** [comment]
running the west build -p auto -b stm32h573i_dk tests/drivers/counter/counter_basic_api/
on all possible counter : Timers 2,3,4,5,6,7,12,13,14 and RTC 
```
------ TESTSUITE SUMMARY START ------                                           
                                                                                
SUITE PASS - 100.00% [counter_basic]: pass = 9, fail = 0, skip = 0, total = 9 ds
 - PASS - [counter_basic.test_all_channels] duration = 3.569 seconds            
 - PASS - [counter_basic.test_cancelled_alarm_does_not_expire] duration = 2.265s
 - PASS - [counter_basic.test_late_alarm] duration = 0.509 seconds              
 - PASS - [counter_basic.test_late_alarm_error] duration = 0.509 seconds        
 - PASS - [counter_basic.test_multiple_alarms] duration = 0.617 seconds         
 - PASS - [counter_basic.test_set_top_value_with_alarm] duration = 0.945 seconds
 - PASS - [counter_basic.test_short_relative_alarm] duration = 6.288 seconds    
 - PASS - [counter_basic.test_single_shot_alarm_notop] duration = 7.649 seconds 
 - PASS - [counter_basic.test_single_shot_alarm_top] duration = 0.649 seconds   
                                                                                
SUITE PASS - 100.00% [counter_no_callback]: pass = 1, fail = 0, skip = 0, totals
 - PASS - [counter_no_callback.test_set_top_value_without_alarm] duration = 0.5s
                                                                                
------ TESTSUITE SUMMARY END ------                        
```

**Comment by FRASTM** [comment]
Running west build -p auto -b stm32h573i_dk tests/drivers/pwm/pwm_loopback/
when D3 is connected to D10 of the arduino connector of the disco kit
```
TESTSUITE pwm_loopback succeeded                                                
                                                                                
------ TESTSUITE SUMMARY START ------                                           
                                                                                
SUITE PASS - 100.00% [pwm_loopback]: pass = 8, fail = 0, skip = 0, total = 8 dus
 - PASS - [pwm_loopback.test_capture_busy] duration = 0.006 seconds             
 - PASS - [pwm_loopback.test_capture_timeout] duration = 1.003 seconds          
 - PASS - [pwm_loopback.test_continuous_capture] duration = 1.200 seconds       
 - PASS - [pwm_loopback.test_period_capture] duration = 0.588 seconds           
 - PASS - [pwm_loopback.test_period_capture_inverted] duration = 0.587 seconds  
 - PASS - [pwm_loopback.test_pulse_and_period_capture] duration = 0.586 seconds 
 - PASS - [pwm_loopback.test_pulse_capture] duration = 0.587 seconds            
 - PASS - [pwm_loopback.test_pulse_capture_inverted] duration = 0.587 seconds   
                                                                                
------ TESTSUITE SUMMARY END ------                  
```

File: tests/drivers/counter/counter_basic_api/boards/stm32h573i_dk.overlay [hunk 1]
```diff
@@ -0,0 +1,71 @@
+&timers2 {
+	st,prescaler = <239>;
+	counter {
+		compatible = "st,stm32-counter";
```
Comment (erwango) [inline tests/drivers/counter/counter_basic_api/boards/stm32h573i_dk.overlay:4]: Remove compatible prop as defined in .dtsi

File: boards/arm/stm32h573i_dk/doc/index.rst [hunk 2]
```diff
@@ -179,6 +179,8 @@ hardware features:
 +-----------+------------+-------------------------------------+
 | ADC       | on-chip    | ADC Controller                      |
 +-----------+------------+-------------------------------------+
+| PWM       | on-chip    | PWM                                 |
++-----------+------------+-------------------------------------+
```
Comment (erwango) [inline boards/arm/stm32h573i_dk/doc/index.rst:183]: Please add RTC as well

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/stm32h573i_dk/doc/index.rst (hunk 2 above)
Comment (FRASTM) [reply boards/arm/stm32h573i_dk/doc/index.rst:183]: Done

**Review by gautierg-st — APPROVED** [review]

File: boards/arm/stm32h573i_dk/doc/index.rst (hunk 2 above)
Comment (erwango) [reply boards/arm/stm32h573i_dk/doc/index.rst:183]: Still missing

**Review by erwango — APPROVED** [review]

**Pull request merged**